src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/SearchService.kt

```diff
@@ -25,146 +26,172 @@ import java.util.*
 
 @Service
 class SearchService @Autowired constructor(private val offenderAccessService: OffenderAccessService, private val hlClient: SearchClient, private val mapper: ObjectMapper) {
-  companion object {
-    val log: Logger = LoggerFactory.getLogger(this::class.java)
-    private const val MAX_SEARCH_RESULTS = 100
-  }
-
-  fun performSearch(searchOptions: SearchDto): List<OffenderDetail> {
-    validateSearchForm(searchOptions)
-    val searchRequest = SearchRequest("offender")
-    val searchSourceBuilder = SearchSourceBuilder()
-    // Set the maximum search result size (the default would otherwise be 10)
-    searchSourceBuilder.size(MAX_SEARCH_RESULTS)
-    val matchingAllFieldsQuery = buildMatchWithAllProvidedParameters(searchOptions)
-    searchSourceBuilder.query(matchingAllFieldsQuery.withDefaults())
-    searchRequest.source(searchSourceBuilder)
-    val response = hlClient.search(searchRequest)
-    return getSearchResult(response)
-  }
-
-  protected fun buildMatchWithAllProvidedParameters(searchOptions: SearchDto): BoolQueryBuilder {
-    val matchingAllFieldsQuery = QueryBuilders
-        .boolQuery()
-    with(searchOptions) {
-      croNumber.takeIf { !it.isNullOrBlank() }?.let {
-        matchingAllFieldsQuery
-            .mustKeyword(it.toLowerCase(), "otherIds.croNumberLowercase")
-      }
-      pncNumber.takeIf { !it.isNullOrBlank() }?.let {
-        matchingAllFieldsQuery
-            .mustMultiMatchKeyword(it.canonicalPNCNumber(), "otherIds.pncNumberLongYear", "otherIds.pncNumberShortYear")
-      }
-      matchingAllFieldsQuery
-          .mustWhenPresent("otherIds.nomsNumber", nomsNumber)
-          .mustWhenPresent("otherIds.crn", crn)
-          .mustWhenPresent("firstName", firstName)
-          .mustWhenPresent("surname", surname)
-          .mustWhenPresent("dateOfBirth", dateOfBirth)
+    companion object {
+        val log: Logger = LoggerFactory.getLogger(this::class.java)
+        private const val MAX_SEARCH_RESULTS = 100
```

Comment:
I think maybe add something to the swagger to indicate this constraint. We might need to raise it for this "bulk" end points, but not sure

Reply:
Ignore, so that doesn't apply to the new endpoints right?

Reply:
I set the max results for this ldu-codes endpoint (searchSourceBuilder.size) to be MAX_SEARCH_RESULTS. Is this appropriate?